composite/src/main/java/com/iluwatar/composite/LetterComposite.java

```diff
@@ -36,18 +36,23 @@
  */
 public abstract class LetterComposite {
 
+  private List<LetterComposite> children = new ArrayList<>();
+
+
   public void add(LetterComposite letter) {
```

Comment:
zur frage b) weshalb das heikel ist: client kann versuchen diese method auf einem Letter durchzuführen (z.B. ein Word zu einem Letter hinzufügen). Das führt dann zu ungültigen Satzstrukturen. Eine mögliche Lösung: add nur im Word und Sentence erlauben:
https://github.com/ibwgr/design-patterns/compare/solution_ex2_2_b_composite

Ein andere: In der Letter-Klasse die Methode überschreiben und nichts tun (oder Fehler werfen)